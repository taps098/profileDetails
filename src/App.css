.App {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100vh;
  background-color: lightgray;
}

.App-logo {
  height: 13vmin;
  width: 13vmin;
  border-radius: 50%;
  pointer-events: none;
}

.user-profile {
  display: flex;
  align-items: center;
  flex-direction: row;
  justify-content: center;
  height:200px;
  width:500px;
  background-color: #fff;
  border-radius: 7px;
}
.user-pic{
  display: flex;
  align-items: center;
  justify-content: center;
  background-image: linear-gradient(rgb(223	166	176)	, rgb(238	205	153	));
  height: 200px;
  width: 250px;
  border-radius: 7px 0 0 7px;
}
.user-details {
  display: flex;
  align-items: flex-start;
  justify-content: center;
  flex-direction: column;
  margin: 12px;
}
.user-name-content{
  display: flex;
  justify-content: space-between;
}
.user-name {
  font-size: large;
  font-weight: 600;
  margin-bottom: 10px;
}
.user-details-content{
  display: flex;
  align-items: flex-start;
  flex-direction: row;
  margin: 5px;
}
.profile-line {
  margin-left: 10px;
  font-size: 12px;
}
.company-logo {
  margin-top: 10px;
  display: flex;
  flex-direction: row;
}
.company-name {
  font-size: 12px;
  font-weight: 500;
}
.company-statement {
  font-weight: 200;
  font-size: 10px;
}
.save-button {
  border: none;
  background-color: #008CBA;
  box-shadow: 1px 1px 3px #888888;
  border-radius: 3px;
  height: 20px;
}
.nameInputStyle {
  border: none;
  box-shadow: 1px 1px 3px #888888;
  border-radius: 3px;
  height: 20px;
  width: 150px;
}

@media all and (max-width: 600px) {
  .user-profile {
    display: flex;
    align-items: center;
    flex-direction: column;
    justify-content: center;
    height:300px;
    width:250px;
    background-color: #fff;
    border-radius: 7px;
  }
  .user-pic {
    display: flex;
    align-items: center;
    justify-content: center;
    background-image: linear-gradient(rgb(223 166 176) , rgb(238 205 153 ));
    height: 100px;
    width: 250px;
    border-radius: 7px 7px 0 0;
}
}