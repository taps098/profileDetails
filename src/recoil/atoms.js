import { atom } from 'recoil';

/**
 * initial value of the recoil state
 */
export const userProfileDetails = atom({
    key: 'userProfileDetails',
    default: {
        userName: "Lois Zemlak",
        mailId: "maida.becker98.gmail.com",
        address: "Clemin Plain st, Apt462, North Connertown, 99373",
        phoneNumber: '(293)4148005',
        org: "marianne.org",
    }
})