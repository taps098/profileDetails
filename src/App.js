import React from 'react';
import UserProfile from './UserProfile';
import './App.css';

/**
 * Function to render the entire component.
 * @returns App
 */
function App() {
  return (
    <div data-testid='app' className="App">
      <UserProfile />
    </div>
  );
}

export default App;
