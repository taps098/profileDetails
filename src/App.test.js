import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import App from './App';
import { RecoilRoot } from "recoil"

test('render the app without fail', () => {
  render(
    <RecoilRoot>
    <App />
    </RecoilRoot>
    );
  const app = screen.getByTestId('app');
  expect(app).toBeInTheDocument();
  expect(app).toMatchSnapshot();
});
test('check for edit button and save button', () => {
  render(
    <RecoilRoot>
    <App />
    </RecoilRoot>
  );
    const editButton = screen.getByTestId('edit-button');
    expect(editButton).toBeInTheDocument();
    fireEvent.click(editButton);
    const saveButton = screen.getByTestId('save-button');
    expect(saveButton).toBeInTheDocument();
});
test('check the the change name function', async () => {
  render(
    <RecoilRoot>
    <App />
    </RecoilRoot>
  );
  const editButton = screen.getByTestId('edit-button');
    expect(editButton).toBeInTheDocument();
    fireEvent.click(editButton);

    // dom should have the input element
    const nameInputField = screen.getByTestId('nameInput');
    fireEvent.change(nameInputField, {target: {value: 'New Name'}})
    const saveButton = screen.getByTestId('save-button');
    expect(saveButton).toBeInTheDocument();
    fireEvent.click(saveButton);
    await waitFor(() => {
      const newNameField = screen.getByTestId('nameField');
      expect(newNameField).toHaveTextContent('New Name');
    }) 
});
