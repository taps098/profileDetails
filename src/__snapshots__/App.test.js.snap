// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the app without fail 1`] = `
<div
  class="App"
  data-testid="app"
>
  <div
    class="user-profile"
  >
    <div
      class="user-pic"
    >
      <img
        alt="logo"
        class="App-logo"
        src="userImage.jpeg"
      />
    </div>
    <div
      class="user-details"
    >
      <div
        class="user-profileDetails"
      >
        <div
          class="user-name-content"
        >
          <div
            class="user-name"
          >
            <div
              data-testid="nameField"
            >
              Lois Zemlak
            </div>
          </div>
          <div>
            <img
              alt="img"
              class="bi bi-pencil"
              data-testid="edit-button"
              src="pencil.svg"
            />
          </div>
        </div>
        <div
          class="user-details-content"
        >
          <img
            alt="img"
            class="bi bi-envelope"
            src="envelope.svg"
          />
          <div
            class="profile-line"
          >
            maida.becker98.gmail.com
          </div>
        </div>
        <div
          class="user-details-content"
        >
          <div>
            <img
              alt="img"
              class="bi bi-file-person"
              src="file-person.svg"
            />
          </div>
          <div
            class="profile-line"
          >
            Clemin Plain st, Apt462, North Connertown, 99373
          </div>
        </div>
        <div
          class="user-details-content"
        >
          <div>
            <img
              alt="img"
              class="bi bi-file-person"
              src="telephone.svg"
            />
          </div>
          <div
            class="profile-line"
          >
            (293)4148005
          </div>
        </div>
        <div
          class="user-details-content"
        >
          <div>
            <img
              alt="img"
              class="bi bi-file-person"
              src="globe.svg"
            />
          </div>
          <div
            class="profile-line"
          >
            marianne.org
          </div>
        </div>
      </div>
      <div
        class="company-logo"
      >
        <div>
          <img
            alt="img"
            class="bi bi-file-person"
            src="bank2.svg"
          />
        </div>
        <div
          class="profile-line"
        >
          <div
            class="company-name"
          >
             McKenzie LLC
          </div>
          <div
            class="company-statment"
          >
            Quality focused value-added synergy
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
